app/controllers/api/v1/skills_controller.rb

```diff
@@ -1,16 +1,24 @@
 class Api::V1::SkillsController < ApplicationController
-
   def create
     begin
       user = User.find(params[:user_id])
 
-      params[:skills].each do |skill|
-        user.skills.create(name: skill[:name], proficiency: skill[:proficiency] )
-        #could we have other error handling here for skills already being present for user? 
+      params[:skills].each do |skill_params|
+        existing_skill = user.skills.find_by(name: skill_params[:name])
+
+        if existing_skill
+          render json: { error: "Skill '#{skill_params[:name]}' already exists for the user." }, status: :unprocessable_entity
```

Comment:
This all checks to see if that skill already exists and if it does it returns the proper message so that there is no duplicates; thank you for adding this!

Reply:
my pleasure